### DISCUSSION TITLE
KEYBUS malfunction

### DISCUSSION THREAD

**Discussion by l33d** [taligentx/dscKeybusInterface#39] [dscKeybusInterface Discussion]
Hi,

Second time i have info od my system about KEYBUS malfunction, before this text i see trouble light and after check the problem in menu *3 i see an "battery error PC520x ( additional power supply?)" I don`t have an additional power supply from dsc in my alarm. Tommorow i check on dls all the problems and paste it. This is second time when i see this text, and keybus error message, after connect virtualblynk.

**Comment by ColinNg** [comment]
If you run the KeybusReader - it will output almost everything it sees to the Serial console. This might be helpful in troubleshooting. If you need to interact with the system, you can send regular keypad commands via the Serial console too.

**Comment by l33d** [comment]
Hi,

I think I've found a problem. By disconnecting and connecting the GND from the ESP that was connected to the KEYBUS, this problem reappeared. I need to connect VCC and GND permanently and see if the problem appears again. If it appears, I will install a sketch keybusreader on second esp and check what it will display. Is the ESP module supervised in the system? I looked in DLS and in the keyboard LCD5500, but did not see an additional module except my 4xpc 5108 1x5208 and 2xLCD5500. Why i see when error occurs battery error in pc520x (2, sometimes i see 1,3,4 too) like on the screen, which i don`t have? 
![error dsc] 

(https://user-images.githubusercontent.com/42523320/44618126-4ad3d400-a870-11e8-9a0d-b4a8bdda1629.PNG)

**Comment by taligentx** [comment]
I'm updating the todo list to note that connections should be soldered and hardwired.  The Keybus data isn't particularly high speed (1Kbps) but any kind of intermittent connectivity can cause problems on the entire Keybus.

For now the interface is not registered as a separate module and will not be visible in DLS.  In theory it is possible to have the interface check for an available slot and register as a keypad - is there any useful function for doing this other than being visible in DLS?

**Comment by l33d** [comment]
How do I understand the interface by itself should not do such problems if the system is actively supervising modules? It's about the situation in which the interface 'pretends' for example an additional power supply, etc. I say that if it suddenly disconnects it, the system stops seeing it and triggers an alarm about KEYBUS od battery error in supply 520X. 

However, if it works now without supervising this ESP  interface and will not make any problems then there is no need to add it.

**Comment by l33d** [comment]
The problem did not return, the cause was probably a loose ground wire.
